Update Valet Key (#159)
mspnp/cloud-design-patterns: valet-key/README.md

Specifically this sample includes an Azure Function that provides a scoped, time-limited shared access signature (SaS) to authorized callers, who would then use that SaS token to perform a data upload to the storage account without consuming the resources of the Azure Function to proxy that request.

> The typical way to generate a SAS token in code requires the storage account key. In this scenario, you won’t have a storage account key, so you’ll need to find another way to generate the shared access signatures. To do that, we need to use an approach called “user delegation” SAS . By using a user delegation SAS, we can sign the signature with the Microsoft Entra ID credentials instead of the storage account key. It is disabled storage account key access.

Reviewer: ```suggestion
The typical way to generate a SAS token in code requires the storage account key. In this scenario, storage account keys are disabled and instead an approach called “user delegation” SAS is used. By using a user delegation SAS, you can sign the signature with Microsoft Entra ID credentials instead of the storage account key.
```

1. Clone this repository to your workstation and navigate to the working directory.

   ```bash
   git clone https://github.com/mspnp/cloud-design-patterns
   cd valet-key
   ```

1. Log into Azure and create an empty resource group.

   ```bash
   az login
   az account set -s <Name or ID of subscription>

>    NAME_PREFIX=valet-key
>    LOCATION=eastus2
>    RESOURCE_GROUP_NAME="rg-${NAME_PREFIX}-${LOCATION}"
> 
>    az group create -n "${RESOURCE_GROUP_NAME}" -l ${LOCATION}

Reviewer: Keep it simple

```suggestion
   LOCATION=eastus2
   RESOURCE_GROUP_NAME="rg-valet-key-${LOCATION}"

   az group create -n "${RESOURCE_GROUP_NAME}" -l ${LOCATION}
```
